pkg/controller/job/job_controller.go

```diff
@@ -880,7 +880,7 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 			}
 			jobCtx.podsWithDelayedDeletionPerIndex = getPodsWithDelayedDeletionPerIndex(logger, jobCtx)
 		}
-		if jobCtx.finishedCondition == nil && hasSuccessCriteriaMetCondition(jobCtx.job) == nil {
```

Comment:
I assumed that there are some situations where the incoming job already has `SuccessCriteriaMet` condition, and we need to verify it. But, it seems that we already add the incoming job's `SuccessCriteriaMet` condition to the `finishedCondition` here https://github.com/kubernetes/kubernetes/blob/5ceb99dc6bae8a3013192439bdbee942d14de190/pkg/controller/job/job_controller.go#L846

(git blame says that L846 was implemented by me :))

Thank you for these cleanups!